packages/ui/src/views/layouts/SandboxRoot.tsx

```diff
@@ -58,58 +34,42 @@ export const SandboxRoot = ({
   pinnedMenu,
   recentMenu,
   logout,
-  changePinnedMenu,
-  changeRecentMenu,
+  setPinned,
+  setRecent,
+  setNavLinks,
   t
 }: SandboxRootProps) => {
   const location = useLocation()
 
-  const [recentMenuItems, setRecentMenuItems] = useState<NavbarItemType[]>([])
-  const [pinnedMenuItems, setPinnedMenuItems] = useState<NavbarItemType[]>([])
   const [showMoreMenu, setShowMoreMenu] = useState(false)
   const [showSettingMenu, setShowSettingMenu] = useState(false)
   const [showCustomNav, setShowCustomNav] = useState(false)
 
   const pinnedMenuItemsData = useMemo(() => getPinnedMenuItemsData(t), [t])
 
-  const handleChangeRecentMenu = useCallback(
-    (nextRouteData: NavbarItemType) => {
-      if (!pinnedMenu || !pinnedMenu.includes(nextRouteData.id)) {
-        changeRecentMenu([...new Set([nextRouteData.id, ...recentMenu])])
-      }
-    },
-    [changeRecentMenu, pinnedMenu, recentMenu]
-  )
+  useLocationChange({ t, onRouteChange: setRecent })
 
-  useLocationChange({ t, onRouteChange: handleChangeRecentMenu })
+  useEffect(() => {
+    const linksFromStorage = localStorage.getItem('nav-items')
```

Comment:
if we are using a zustand store for saving pinned items, adding a `persist` middle should give us this functionality

Reply:
This is to set the very first time pinned column links for the user. As the pinned column data is in `SandboxRoot` component, we cannot set the zustand state directly in `RootWrapper`. So setting the initial pinned links this way.